connector/src/main/java/com/google/cloud/hive/bigquery/connector/input/avro/AvroSerializer.java

```diff
@@ -145,7 +144,7 @@ public static Object serialize(
 
     if (objectInspector instanceof HiveDecimalObjectInspector) {
       byte[] bytes = ((ByteBuffer) avroObject).array();
-      int scale = (int) actualSchema.getObjectProp("scale");
+      int scale = actualSchema.getJsonProp("scale").asInt();
```

Comment:
this getJsonProp won't work with newer avro versions? in avro-1.11 it is a private method

Reply:
I've added a workaround for this in the latest commit. Please take a look